packages/app/src/components/ErrorPages/illustrations/illustration.tsx

```diff
@@ -0,0 +1,28 @@
+import Box, { BoxProps } from '@mui/material/Box';
+import { useTheme } from '@mui/material/styles';
+
+export const createIllustration = (
+  light: string,
+  dark: string,
+  displayName?: string,
+) => {
+  const Illustration = (props: BoxProps<'img'>) => {
+    const theme = useTheme();
+    const isDarkMode = theme.palette.mode === 'dark';
+
+    return (
+      <Box
+        component="img"
+        src={isDarkMode ? dark : light}
+        // presentational image
+        alt=""
+        aria-hidden="true"
+        {...props}
+      />
+    );
+  };
+
+  Illustration.displayName = displayName || 'Illustration';
```

Comment:
It's not a big issue, but this `displayName` arg and need to set a displayName is a hint for me, that the composition order could be easier (in the other direction).

Like:

NotFoundErrorPage use CollaborationIllustration which could use Illustration with two props.

But I don't want delay this PR more, so let us merge it, its not wrong and actually "just" used on an error page.